src/pip/_internal/req/req_install.py

```diff
@@ -108,7 +108,13 @@ def __init__(
             # PEP 508 URL requirement
             link = Link(req.url)
         self.link = self.original_link = link
-        self.original_link_is_in_wheel_cache = False
+
+        # When is_wheel_from_cache is True, it means that this InstallRequirement
+        # is a local wheel file in the cache of locally built wheels.
+        self.is_wheel_from_cache = False
+        # When is_wheel_from_cache is True, this is the source link corresponding
+        # to the cache entry, which was used to download and build the cached wheel.
+        self.cached_wheel_source_link: Optional[Link] = None
```

Comment:
Do we need two attributes for this? From what I can tell these are always set together and `is_wheel_from_cache` can be entirely implied by `cached_wheel_source_link is not None`.

Reply:
This is indeed a logical next step in this refactoring. Done.